src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -183,6 +185,50 @@ Scalar<DataVector> area_element(
            square(get(dot_product(cap_theta, cap_phi, spatial_metric))));
   return area_element;
 }
+
+template <typename Frame>
+Scalar<DataVector> spin_function(
+    const StrahlkorperTags::StrahlkorperTags_detail::Jacobian<Frame>& tangents,
+    const YlmSpherepack& ylm,
+    const tnsr::I<DataVector, 3, Frame>& unit_normal_vector,
+    const Scalar<DataVector>& area_element,
+    const tnsr::ii<DataVector, 3, Frame>& extrinsic_curvature) noexcept {
+  auto sin_theta = make_with_value<Scalar<DataVector>>(area_element, 0.0);
+  get(sin_theta) = sin(ylm.theta_phi_points()[0]);
+
+  auto extrinsic_curvature_theta_normal_sin_theta =
+      make_with_value<Scalar<DataVector>>(area_element, 0.0);
+  auto extrinsic_curvature_phi_normal =
+      make_with_value<Scalar<DataVector>>(area_element, 0.0);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      // Note: I must multiply by sin_theta here because
+      // I take the phi derivative of this term by using
+      // the spherepack gradient, which includes a
+      // sin_theta in the denominator of the phi derivative.
+      get(extrinsic_curvature_theta_normal_sin_theta) +=
```

Comment:
precompute K_{ij} s^j sin(theta) as you are multiplying both tangents by the same factor